test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -397,6 +398,20 @@ def closure():
         # Load the optimizer state dict, check that no exception is raised
         optimizer.load_state_dict(optimizer_state_dict)
 
+        # Check that the states are not None, but {}
```

Comment:
this is a new failsafe test, the previous iteration would None the parameters in the state dict not belonging to this rank, and lightning would fail on that. Making sure that this does not reappear